src/mlpack/methods/bayesian_ridge/bayesian_ridge_main.cpp

```diff
@@ -0,0 +1,185 @@
+    "in more detail below:"
+    "\n\n"
+    "Let X be a matrix where each row is a point and each column is a "
+    "dimension, t is a vector of targets, alpha is the precision of the "
+    "gaussian prior distribtion of w, and w is solution to determine. "
+    "\n\n"
+    "The Bayesian Ridge comptutes the posterior distribution of the parameters "
+    "by the Bayes's rule : "
+    "\n\n"
+    " p(w|X) = p(X,t|w) * p(w|alpha) / p(X)"
+    "\n\n"
+    "To train a BayesianRidge model, the " +
+    PRINT_PARAM_STRING("input") + " and " + PRINT_PARAM_STRING("responses") +
+    "parameters must be given. The " + PRINT_PARAM_STRING("center") +
+    "and " + PRINT_PARAM_STRING("scale") + " parameters control the "
+    "centering and the normalizing options. A trained model can be saved with "
+    "the " + PRINT_PARAM_STRING("output_model") + ". If no training is desired "
+    "at all, a model can be passed via the "+ PRINT_PARAM_STRING("input_model")+
+    " parameter."
+    "\n\n"
+    "The program can also provide predictions for test data using either the "
+    "trained model or the given input model.  Test points can be specified with"
+    " the " + PRINT_PARAM_STRING("test") + " parameter.  Predicted responses "
+    "to the test points can be saved with the " +
+    PRINT_PARAM_STRING("output_predictions") + " output parameter."
+    "\n\n"
+    "For example, the following command trains a model on the data " +
+    PRINT_DATASET("data") + " and responses " + PRINT_DATASET("responses") +
+    "with center set to true and scale set to false (so, Bayesian "
+    "Ridge is being solved, and then the model is saved to " +
+    PRINT_MODEL("bayesian_ridge_model") + ":"
+    "\n\n" +
+    PRINT_CALL("bayesian_ridge", "input", "data", "responses", "responses",
+               "center", 1, "scale", 0, "output_model",
+               "bayesian_ridge_model") +
+    "\n\n"
+    "The following command uses the " + PRINT_MODEL("bayesian_ridge_model") +
+    " to provide predicted responses for the data " + PRINT_DATASET("test") +
+    " and save those responses to " + PRINT_DATASET("test_predictions") + ": "
+    "\n\n" +
+    PRINT_CALL("bayesian_ridge", "input_model", "bayesian_ridge_model", "test",
+               "test", "output_predictions", "test_predictions"));
+
+PARAM_TMATRIX_IN("input", "Matrix of covariates (X).", "i");
+
+PARAM_MATRIX_IN("responses", "Matrix of responses/observations (y).", "r");
+
+PARAM_MODEL_IN(BayesianRidge, "input_model", "Trained BayesianRidge model "
+               "to use.", "m");
+
+PARAM_MODEL_OUT(BayesianRidge, "output_model", "Output BayesianRidge model.",
+                "M");
+
+PARAM_TMATRIX_IN("test", "Matrix containing points to regress on (test "
+                 "points).", "t");
+
+PARAM_TMATRIX_OUT("output_predictions", "If --test_file is specified, this "
+                  "file is where the predicted responses will be saved.", "o");
```

Comment:
Do you think it would be useful to add an output parameter for the standard deviations of each prediction, too?

You can also set the parameter type here to `PARAM_ROW_OUT` and then the `.t()` won't be necessary when calling `CLI::GetParam<...>()` to set its value at the bottom of the program. :+1:

Reply:
Yes, of course ! I added it.